torch_xla/experimental/distributed_checkpoint/manager.py

```diff
@@ -139,10 +188,18 @@ def restore(self, step: int, state_dict: STATE_DICT_TYPE) -> None:
       state_dict: The state dict to restore the checkpoint into. Values are
                   updated in-place within the state_dict.
     """
-    raise NotImplementedError
+    path = self._get_path(step)
+    dist_cp.load_state_dict(
+        state_dict=state_dict,
+        storage_reader=FsspecReader(path),
+        planner=xc.SPMDLoadPlanner(),
+    )
 
   def all_steps(self) -> List[int]:
     """
     List all steps tracked by the CheckpointManager.
     """
-    raise NotImplementedError
+    fs, raw_path = url_to_fs(self.base_path)
```

Comment:
What's this? Why don't we just keep a queue as a private member?

Reply:
`url_to_fs` takes in a path like `gs://my-bucket/path/to/chkpt` and returns an fsspec FileSystem object and a raw path without the storage prefix, e.g. `my-bucket/path/to/chkpt`. The FileSystem object is what's used to interface with the storage backend.

I'd prefer to read from the source-of-truth each time in case there are any changes on the backend, but we could make this a one-time operation and track a queue within the manager. Let me know what you prefer.

Reply:
I see. It depends on how hot this path will be? If it's a hot path, it'd better cached in the manager. Otherwise, it seems okay.

Reply:
It will be called once per checkpoint, but we may as well cache it.